src/app/backend/resource/container/logs.go

```diff
@@ -117,8 +112,31 @@ func getRawPodLogs(client kubernetes.Interface, namespace, podID string, logOpti
 	return string(result), nil
 }
 
-// Build logs structure for given parameters.
-func ConstructLogs(podID string, rawLogs string, container string, logSelector *logs.Selection) *logs.LogDetails {
+// GetLogFile returns a stream to the log file which can be piped directly to the respose. This avoids oom issues.
+func GetLogFile(client kubernetes.Interface, namespace, podID string, container string) (string, io.ReadCloser, error) {
+	logOptions := &v1.PodLogOptions{
+		Container:  container,
+		Follow:     false,
+		Previous:   false,
+		Timestamps: false,
+	}
+	filename := fmt.Sprintf("logs-from-%v-in-%v.txt", container, podID)
+	logStream, err := openStream(client, namespace, podID, logOptions)
+	return filename, logStream, err
+}
+
+func openStream(client kubernetes.Interface, namespace, podID string, logOptions *v1.PodLogOptions) (io.ReadCloser, error) {
+	req := client.CoreV1().RESTClient().Get().
```

Comment:
Any reason to create this temp value? Can't `Stream()` be chained with previous line?

Reply:
chained it